### PR TITLE
Ek tests

### PR DESCRIPTION
**Pull request by ekaspern**
did the best i could :)

### CODE REVIEW

File: src/components/pvr.coffee [hunk 1]
```diff
@@ -200,6 +207,8 @@ Pvr = React.createClass
 
     direction = @calculateOptimalDirection elPos
 
+    console.log "direction", direction
```
Comment (darcyadams) [inline src/components/pvr.coffee]: log statement

File: tests/components/input_placeholder.coffee [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    inputPlaceHolder = TestUtils.renderIntoDocument InputPlaceHolder {
+      onChange: onChange
+      value: 'text'
+      placeholder: 'placeholder text'
+      placeholderIsSupported: false
+    }
+
+    placeHolderClass = TestUtils.scryRenderedDOMComponentsWithClass inputPlaceHolder, 'placeholder'
+   
+    expect(placeHolderClass.length).to.equal(0)
```
Comment (darcyadams) [inline tests/components/input_placeholder.coffee]: maybe another test here that it adds the placeholder text when `placeholderIsSupported: false` and the value is `''`

File: tests/components/input_type_ahead.coffee [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+      }
+   
+    inputText = TestUtils.findRenderedDOMComponentWithTag inputTypeAhead, 'input'
+
+    TestUtils.Simulate.change inputText
+
+    expect(onChange.called).to.equal(true)
+
+  #--------------------------------------------------------------------- Display Div
+  it 'Should display No Results if no results array is empty', ->
```
Comment (darcyadams) [inline tests/components/input_type_ahead.coffee]: 'Should display No Results if ~~no~~ results array is empty'

File: tests/components/pvr.coffee [hunk 4]
```diff
@@ -0,0 +1,439 @@
...
+          left: 400
+          right: 500  
+      }
+    }
+
+    pvrMenu = TestUtils.findRenderedDOMComponentWithClass pvr, 'pvr'
+
+    expect(pvrMenu.className).to.contain('left')
+
+  it 'Should have class of right if no direction set and room for menu on right', ->
```
Comment (darcyadams) [inline tests/components/pvr.coffee]: `Should have a class of right if no direction is set, there is no room on the left, but there is room for the menu on the right'? - no need for brevity, these are supposed to read like sentences.

File: tests/components/pvr.coffee [hunk 5]
```diff
@@ -0,0 +1,439 @@
...
+      stopPropagation: sinon.spy()
+
+    TestUtils.Simulate.click el, e
+
+    expect(e.stopPropagation.called).to.equal(true)
+
+
+  #--------------------------------------------------------------------- Position No Direction 
+  #--------------------------------------------------------------------- expects window size 300 X 400
+  it 'Should have class of left if no direction set and room for menu on left', ->
```
Comment (darcyadams) [inline tests/components/pvr.coffee]: would like to see some comment here explaining the math on the dimensions, ie, how big is the window? why are you setting the `getBoundingClientRect` measurements to these values? anything that will make troubleshooting these tests easier in six months.

Comment (darcyadams) [inline tests/components/pvr.coffee]: also, 
'Should have class of left if no direction set and room for menu on left'
doesn't seem to fully describe the conditions that lead to a left direction, isn't left the last direction used when no direction is passed?

Comment (darcyadams) [inline tests/components/pvr.coffee]: sorry, missed this comment

File: tests/components/pvr.coffee [hunk 6]
```diff
@@ -0,0 +1,439 @@
...
+          right: 200  
+      }
+    }
+
+    pvrMenu = TestUtils.findRenderedDOMComponentWithClass pvr, 'pvr'
+
+    expect(pvrMenu.className).to.contain('below')
+
+
+  it 'Should have class of above if direction is set to above and room above', ->
```
Comment (darcyadams) [inline tests/components/pvr.coffee]: I feel like this should be the first test in each section, and subsequent tests move on to the exception cases.

File: tests/components/pvr.coffee [hunk 7]
```diff
@@ -0,0 +1,439 @@
...
+          bottom: 55
+          left: 10
+          right: 10  
+      }
+    }
+
+    pvrMenu = TestUtils.findRenderedDOMComponentWithClass pvr, 'pvr'
+
+    expect(pvrMenu.className).to.contain('below')
+
```
Comment (darcyadams) [inline tests/components/pvr.coffee]: seems like there are some other features that could be tested here:
hSlide
vSlide
hAdjust
vAdjust
Is the position set correctly based on the anchor element
height
width
styleMixin
etc

File: tests/components/input_placeholder.coffee (hunk 2 above)
Comment (ekaspern) [inline tests/components/input_placeholder.coffee]: yup, we have this test. it is above the other one. could change the wording?

**Pull request merged**